pkg/apis/core/types.go

```diff
@@ -3988,7 +3988,10 @@ type ServiceAccount struct {
 	// +optional
 	metav1.ObjectMeta
 
-	// Secrets is the list of secrets allowed to be used by pods running using this ServiceAccount
+	// Secrets is a list of the secrets in the same namespace that pods running using this ServiceAccount are allowed to use.
+	// Pods are only limited to this list if this service account has a "kubernetes.io/enforce-mountable-secrets" annotation set to "true".
```

Comment:
once `LimitSecretReferences` is `true`, this will be no-op. maybe update the TODO https://github.com/kubernetes/kubernetes/blob/b917653296b6fcac2220afb3ef4f8ed50a2025a2/plugin/pkg/admission/serviceaccount/admission.go#L104 to remember to update the doc.

Reply:
there are no plans to set LimitSecretReferences to true